stat/stat.go

```diff
@@ -217,6 +217,35 @@ func Correlation(x, y, weights []float64) float64 {
 	return (sxy - xcompensation*ycompensation/sumWeights) / math.Sqrt(sxx*syy)
 }
 
+// KendallCorrelation returns the weighted Kendall correlation between the
+// samples of x and y.
+// The lengths of x and y must be equal.
+func KendallCorrelation(x, y []float64) float64 {
```

Comment:
The comment says that it's a weighted Kendall correlation, but this doesn't have weights. My impression is that this test doesn't make sense with weighted data, but I'm not an expert in this test. Do you agree @sbinet @kortschak @jonlawlor ?

Reply:
@Armadilloa16 Do you have some time for this?

I found [this](http://au.mathworks.com/matlabcentral/fileexchange/27361-weighted-kendall-rank-correlation-matrix) that suggests a weighted form is possible, but I don't know enough about it to be able to comment otherwise. Also [this](http://www.sciencedirect.com/science/article/pii/S0167715298000066) - apologies for anyone not at an institution.